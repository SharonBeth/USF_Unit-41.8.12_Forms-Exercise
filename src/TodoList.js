import './App.css';
import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

function TodoList() {
    const [todos, setTodos] = useState([]);
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, { ...newTodo, id: uuid() }])
    }
    const update = (id, updateTodo) => {
        setTodos(todos.map(todo => todo.id === id ? { ...todo, task: updateTodo } : todo))
    }
    const remove = id => {
        setTodos(todos => todos.filter(todo => todo.id !== id));
    };

    return (
        <div className="TodoList">
            <h3> Todo List</h3>
            <NewTodoForm addTodo={addTodo} />
            <div>
                {todos.map(todo => (<Todo id={todo.id} key={todo.id} task={todo.task} update={update} handleRemove={remove} />))}
            </div>
        </div>
    )
}

export default TodoList;