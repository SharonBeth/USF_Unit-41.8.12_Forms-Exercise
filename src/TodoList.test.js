import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing-Smoke Test", function () {
    render(<TodoList />)
});

// it("passes the snapshot test", function () {
// const { asFragment } = render(<TodoList />)
// expect(asFragment()).toMatchSnapshot();
// });

it("should add a todo Item", function () {
    const { queryByText, getByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText("Task:")
    const btn = queryByText("Add a todo!")
    expect(getByText("feed cats")).not.toBeInTheDocument();

    fireEvent.change(input, { target: { value: "feed cats" } });
    fireEvent.click(btn);
    const { asFragment } = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot();

    expect(queryByText("Task:feed cats")).toBeInTheDocument();
})