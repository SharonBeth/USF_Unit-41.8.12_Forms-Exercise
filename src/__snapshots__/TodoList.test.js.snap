// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a todo Item 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h3>
       Todo List
    </h3>
    <div>
      <form>
        <label
          for="task"
        >
          Task:
        </label>
        <input
          id="task"
          name="task"
          type="text"
          value=""
        />
        <button>
          Add a todo!
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
